camel/utils/token_counting.py

```diff
@@ -65,6 +65,27 @@ def messages_to_prompt(messages: List[OpenAIMessage], model: ModelType) -> str:
             else:
                 ret += role + ":"
         return ret
+    elif model.is_anthropic:
+        from anthropic import AI_PROMPT, HUMAN_PROMPT
+
+        # use XML tag to decorate prompt
+        # https://docs.anthropic.com/claude/docs/constructing-a-prompt#mark-different-parts-of-the-prompt
+        # https://docs.anthropic.com/claude/docs/roleplay-dialogue
+        # https://docs.anthropic.com/claude/docs/human-and-assistant-formatting
+        system_prompt = f"\n{system_message}\n"
+        ret = system_prompt
+        for msg in messages[1:]:
+            role, content = msg["role"], msg["content"]
+            # Claude does not perform well if the system message RULE OF USER/ASSISTANT is sent twice. (see role_playing/RolePlaying.init_chat())
```

Comment:
Thank you for spotting. Did you discover a bug of sending the system prompt twice? We never observed it. Could you create a bug ticket to reproduce it and fix? We'd better fix the erroneous behavior in the main code than create a detour here in your code.

Reply:
I think it is an expected behavior since in init_chat() the doc_string says "..sending the system message again to the agents using chat messages". You can spot it [here](https://github.com/camel-ai/camel/blob/3ba175450085b03d69ac1c68156c41d3ae843204/camel/societies/role_playing.py#L351C4-L351C4). 
I should say it is confusing for me since in `role_play`, this `init_chat` function is used to get the first `input_assistant_msg` and sometimes it is just a modified version of `user_sys_msg` (for example in the [open source example](https://github.com/camel-ai/camel/blob/3ba175450085b03d69ac1c68156c41d3ae843204/examples/open_source_models/role_playing_with_open_source_model.py#L60C9-L60C9)). I'm happy to help if you can confirm this is not desired.